apps/hash-frontend/src/pages/flows.page/flow-definition.tsx

```diff
@@ -305,16 +318,66 @@ export const FlowDefinition = () => {
     [flowRuns, selectedFlow.name],
   );
 
+  const [startFlow] = useMutation<
+    StartFlowMutation,
+    StartFlowMutationVariables
+  >(startFlowMutation);
+
+  const handleRunFlowClicked = () => {
+    if (selectedFlow.trigger.outputs?.length) {
+      setShowRunModal(true);
+    } else {
+      void startFlow({
+        variables: {
+          flowDefinition: selectedFlow,
+          flowTrigger: {
+            triggerDefinitionId: "userTrigger",
+          },
+        },
+      });
+    }
+  };
+
+  const flowDefinitionStateKey = `${selectedFlow.name}`;
+  const flowRunStateKey = `${flowDefinitionStateKey}-${selectedFlowRun?.runId ?? "definition"}`;
+
   return (
     <Box
       sx={{
+        height: "calc(100vh - 60px)",
         width: "100%",
-        background: ({ palette }) => palette.gray[10],
+        background: ({ palette }) =>
+          selectedFlowRun ? palette.gray[10] : "rgb(241, 246, 251)",
         pb: 8,
+        transition: ({ transitions }) =>
+          transitions.create("background", transitionOptions),
       }}
     >
+      <RunFlowModal
+        key={selectedFlow.name}
+        flowDefinition={selectedFlow}
+        open={showRunModal}
+        onClose={() => setShowRunModal(false)}
+        runFlow={(outputs: FlowTrigger["outputs"]) => {
+          void startFlow({
+            variables: {
+              flowDefinition: selectedFlow,
+              flowTrigger: {
+                outputs,
+                triggerDefinitionId: "userTrigger",
+              },
+            },
+          });
+          setShowRunModal(false);
+        }}
+      />
```

Comment:
Doesn't block merging, but it'd be nice if the flow run that has been started comes into view automatically.

Reply:
We can do this, but it'll involve amending the `runFlow` mutation to just return the id rather than wait for execution to complete. Are you happy for me to do this? I didn't know if you were still using it somehow for testing

Reply:
Ah I see, I am still using the GQL endpoint directly for testing and find it useful to get the outputs there.

Perhaps we can add an optional `returnOutputs` parameter or something like that, to allow the caller to decide?

Reply:
I wouldn't bother, I don't think we really need it to return the outputs. In the GraphQL playground you could trigger a flow run and then switch to running `getFlowRuns` to get the output (optionally filtered to just that one's id). But we can leave it how it is for a bit longer

Reply:
Alright, fair enough